spec/commitments_deprecater_spec.rb

```diff
@@ -0,0 +1,197 @@
+  end
+
+  def lookup_commitment(ocn)
+    Cluster.where(ocns: ocn).first.commitments.first
+  end
+
+  before(:each) do
+    Cluster.collection.find.delete_many
+    dep.clear_err
+  end
+
+  # Some of the tests below come in pairs, with one pos(itive) and neg(ative) per method.
+  # Pos = where we expect to find something and we do
+  # Neg = where we expect to find nothing and we don't
+
+  it "logs to a file" do
+    expect(dep.log_path).to eq nil
+    dep.log("foo")
+    expect(File.file?(dep.log_path)).to be true
+  end
+
+  it "check header, neg" do
+    expect(dep.check_header("")).to be false
+    expect(dep.check_header("foo")).to be false
+    expect(dep.err.size).to eq 2
+  end
+
+  it "check header, pos" do
+    expect(dep.check_header("organization\tocn\tlocal_id\tdeprecation_status")).to be true
+    expect(dep.err.size).to eq 0
+  end
+
+  it "check cluster, pos" do
+    expect(dep.check_cluster(empty_cluster.ocns.first)).to be_a Cluster
+    expect(dep.err.size).to eq 0
+  end
+
+  it "check cluster, neg" do
+    expect(dep.check_cluster(-999)).to eq nil
+    expect(dep.err).to eq ["No cluster found for OCN:-999"]
+  end
+
+  it "check_commitments, pos" do
+    comm = make_commitment(ocn_1, org_1, local_id_1)
+    cluster = comm.cluster
+    expect(dep.check_commitments(cluster).size).to eq 1
+    expect(dep.check_commitments(cluster).first).to be_a Clusterable::Commitment
+    expect(dep.err.size).to eq 0
+  end
+
+  it "check_commitments, neg" do
+    expect(dep.check_commitments(empty_cluster).size).to eq 0
+    expect(dep.err.size).to eq 0
+  end
+
+  it "check_commitments_by_org, pos" do
+    comm = make_commitment(ocn_1, org_1, local_id_1)
+    ret = dep.check_commitments_by_org([comm], org_1)
+    expect(ret).to eq [comm]
+    expect(dep.err.size).to eq 0
+  end
+
+  it "check_commitments_by_org, neg" do
+    comm = make_commitment(ocn_1, org_1, local_id_1)
+    ret = dep.check_commitments_by_org([comm], org_2)
+    expect(ret.size).to eq 0
+    expect(dep.err).to eq ["No commitments by organization:#{org_2} in cluster."]
+  end
+
+  it "reject_deprecated rejects deprecated" do
+    com1 = make_commitment(ocn_1, org_1, local_id_1)
+    com2 = make_commitment(ocn_1, org_1, local_id_2)
+    com2.deprecate(status: "E")
+    expect(com1.deprecated?).to be false
+    expect(com2.deprecated?).to be true
+    ret = dep.reject_deprecated([com1, com2])
+    expect(ret).to eq [com1]
+  end
+
+  it "reject_deprecated _only_ rejects deprecated" do
+    com1 = make_commitment(ocn_1, org_1, local_id_1)
+    com2 = make_commitment(ocn_1, org_1, local_id_2)
+    ret = dep.reject_deprecated([com1, com2])
+    expect(ret).to eq [com1, com2]
+  end
+
+  it "check_local_id_matches" do
+    ret = dep.check_local_id_matches(
+      [Mockmitment.new(local_id_1), Mockmitment.new(local_id_2)],
+      local_id_1
+    )
+    expect(ret.size).to eq 1
+    expect(ret.first.local_id).to eq local_id_1
+  end
+
+  it "check_multiple_matches" do
```

Comment:
e.g.
`describe "#check_multiple_matches" do`
`   it "finds duplicate local_ids" do` # or whatever this does
`     expect...`
`   end`
`end`